src/DynamoCore/Core/CustomNodeManager.cs

```diff
@@ -241,10 +241,15 @@ protected virtual void OnCustomNodeRemoved(Guid functionId)
 
             CustomNodeWorkspaceModel workspace = null;
             if (loadedWorkspaceModels.TryGetValue(id, out workspace))
+            {
                 RegisterCustomNodeInstanceForUpdates(node, workspace);
+                // Check for invalid input symbols and do an intial definition sync.
+                var removeErrorState = ValidateInputSymbolNames(node, workspace);
```

Comment:
Can you set the node state to be in an error/warning state using `node.State = ...` and later check for the `State` property on the node in `ValidateDefinition`? This way you'll not have to pass the boolean flag `removeErrorState` into functions `ResyncWithDefinition` and `ValidateDefinition` and then not have to break API's.